manager/dispatcher/assignments.go

```diff
@@ -34,9 +34,9 @@ func newAssignmentSet(log *logrus.Entry, dp *drivers.DriverProvider) *assignment
 	}
 }
 
-func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID) {
+func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID, serviceID string) {
```

Comment:
> The task has a copy of the TaskSpec. Anything taken from service.Spec.Task should be taken from task.Spec instead, to avoid this potential inconsistency. Same with task.Endpoint vs. service.Endpoint.

In that case should this function just take the task instead of the service ID?  It looks like we won't actually have to look up the service ID and get the service, since the `driver.Get` function requires just the container hostname, the service annotations, and the service endpoint, which can be gotten from:

- `task.Spec.GetContainer().Hostname` (if the container isn't nil)
- `task.ServiceAnnotations`
- `task.Endpoint`